### ISSUE TITLE
VS Code corrupts multiline BASH_FUNC variables when started from wsl

### ISSUE THREAD

**Issue by bencardinal** [microsoft/vscode-remote-release#9457]
Type: <b>Bug</b>

1. Start a WSL distro created from a `ubi8-minimal` Docker export
1. Launch VS code from the WSL terminal prompt by running `code .`
1. The following error is printed when the VS code integrated terminal opens:
    ```
    bash: switchml: line 1: syntax error: unexpected end of file
    bash: error importing function definition for `switchml'
    bash: _module_raw: line 1: syntax error: unexpected end of file
    bash: error importing function definition for `_module_raw'
    /bin/sh: switchml: line 1: syntax error: unexpected end of file
    /bin/sh: error importing function definition for `switchml'
    /bin/sh: _module_raw: line 1: syntax error: unexpected end of file
    /bin/sh: error importing function definition for `_module_raw'
    /bin/sh: switchml: line 1: syntax error: unexpected end of file
    /bin/sh: error importing function definition for `switchml'
    /bin/sh: _module_raw: line 1: syntax error: unexpected end of file
    /bin/sh: error importing function definition for `_module_raw'
    ```
1. The symptoms are very similar to issues repored by these two (closed) issues:
  - https://github.com/microsoft/vscode/issues/163564
  - https://github.com/microsoft/vscode/issues/3928
1. Running `printenv | grep BASH_FUNC` shows duplicate entires for `BASH_FUNC_switchml%%`
    ```
    BASH_FUNC_switchml%%=() {  typeset swfound=1;
    BASH_FUNC__module_raw%%=() {  unset _mlshdbg;
    BASH_FUNC_which%%=() {  ( alias;
    BASH_FUNC_module%%=() {  _module_raw "$@" 2>&1
    BASH_FUNC__module_raw%%=() {  unset _mlshdbg;
    BASH_FUNC_switchml%%=() {  typeset swfound=1;
    BASH_FUNC_scl%%=() {  if [ "$1" = "load" -o "$1" = "unload" ]; then
    BASH_FUNC_ml%%=() {  module ml "$@"
    ```
1. The first entry for `BASH_FUNC_switchml%%` is only the first line while the second entry is for the entire multiline function as seen by running `printenv BASH_FUNC_switchml%%`:
    ```
    () {  typeset swfound=1;
    () {  typeset swfound=1;
     if [ "${MODULES_USE_COMPAT_VERSION:-0}" = '1' ]; then
     typeset swname='main';
     if [ -e /usr/share/Modules/libexec/modulecmd.tcl ]; then
     typeset swfound=0;
     unset MODULES_USE_COMPAT_VERSION;
     fi;
     else
     typeset swname='compatibility';
     if [ -e /usr/share/Modules/libexec/modulecmd-compat ]; then
     typeset swfound=0;
     MODULES_USE_COMPAT_VERSION=1;
     export MODULES_USE_COMPAT_VERSION;
     fi;
     fi;
     if [ $swfound -eq 0 ]; then
     echo "Switching to Modules $swname version";
     source /usr/share/Modules/init/bash;
     else
     echo "Cannot switch to Modules $swname version, command not found";
     return 1;
     fi
    }
    ```
1. If I invoke `code .` for the same path from within powershell and then reopen in wsl from within VS code (as prompted) there are no issues.

Extension version: 0.81.8
VS Code version: Code 1.81.1 (6c3e3dba23e8fadc360aed75ce363ba185c49794, 2023-08-09T22:22:42.175Z)
OS version: Windows_NT x64 10.0.19045
Modes:
Remote OS version: Linux x64 5.15.133.1-microsoft-standard-WSL2

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|13th Gen Intel(R) Core(TM) i7-13850HX (28 x 2304)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|63.69GB (31.55GB free)|
|Process Argv|--folder-uri=vscode-remote://wsl+simcore-dev/home/simdev --remote=wsl+simcore-dev|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: simcore-dev|
|OS|Linux x64 5.15.133.1-microsoft-standard-WSL2|
|CPUs|13th Gen Intel(R) Core(TM) i7-13850HX (20 x 2303)|
|Memory (System)|31.19GB (28.57GB free)|
|VM|0%|
</details>

**Comment by shareefj** [comment]
+1

**Comment by aeschli** [comment]
Fixed in 0.88.5

**Comment by aeschli** [comment]
Duplicate of https://github.com/microsoft/vscode/issues/206882
